### ISSUE TITLE
Uncaught SyntaxError: Expected "(" or name but "^" found.

### ISSUE THREAD

**Issue by shcherbak** [mulesoft/api-workbench#300]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.9.9
**System**: "Fedora release
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.36

### Stack Trace

Uncaught SyntaxError: Expected "(" or name but "^" found.

```
At /home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/node_modules/raml-definition-system/node_modules/raml-typesystem/dist/src/typeExpressionParser.js:680

SyntaxError: Expected "(" or name but "^" found.
    at peg$buildException (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/node_modules/raml-definition-system/node_modules/raml-typesystem/dist/src/typeExpressionParser.js:166:20)
    at Object.peg$parse [as parse] (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/node_modules/raml-definition-system/node_modules/raml-typesystem/dist/src/typeExpressionParser.js:680:19)
    at Object.parse (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/node_modules/raml-definition-system/node_modules/raml-typesystem/dist/src/typeExpressions.js:126:27)
    at Object.parse (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/node_modules/raml-definition-system/node_modules/raml-typesystem/dist/src/typeExpressionUtil.js:12:15)
    at ReferencePatcher.patchType (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/referencePatcher.js:177:52)
    at ReferencePatcher.patchReferences (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/referencePatcher.js:62:18)
    at ReferencePatcher.patchReferences (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/referencePatcher.js:69:18)
    at ReferencePatcher.patchReferences (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/referencePatcher.js:69:18)
    at ReferencePatcher.process (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/referencePatcher.js:20:14)
    at TraitsAndResourceTypesExpander.expandTraitsAndResourceTypes (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/expander.js:90:53)
    at Object.expandTraitsAndResourceTypes (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/expander.js:26:10)
    at ASTNodeImpl.validate (/home/zip/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/highLevelImpl.js:920:35)
    at c (/home/zip/.atom/packages/api-workbench/main.js:2:5468)
    at u (/home/zip/.atom/packages/api-workbench/main.js:2:5006)
    at p (/home/zip/.atom/packages/api-workbench/main.js:2:4339)
    at y (/home/zip/.atom/packages/api-workbench/main.js:2:5768)
    at Workspace.module.exports.Workspace.observeTextEditors (/usr/share/atom/resources/app.asar/src/workspace.js:330:9)
    at Object.s [as initEditorObservers] (/home/zip/.atom/packages/api-workbench/main.js:2:4061)
    at Object.v [as consumeLinter] (/home/zip/.atom/packages/api-workbench/main.js:1:1208)
    at Provider.module.exports.Provider.provide (/usr/share/atom/resources/app.asar/node_modules/service-hub/lib/provider.js:30:52)
    at /usr/share/atom/resources/app.asar/node_modules/service-hub/lib/service-hub.js:55:26
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

### Commands

```
```

### Config

```json
{}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.36 (inactive)
linter, v1.11.16 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.51.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.43.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.3.2 (active)
one-light-ui, v1.3.2 (inactive)
one-dark-syntax, v1.2.0 (active)
one-light-syntax, v1.2.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.2 (inactive)
archive-view, v0.61.1 (inactive)
autocomplete-atom-api, v0.10.0 (inactive)
autocomplete-css, v0.11.1 (inactive)
autocomplete-html, v0.7.2 (inactive)
autocomplete-plus, v2.31.0 (inactive)
autocomplete-snippets, v1.11.0 (inactive)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (inactive)
background-tips, v0.26.0 (inactive)
bookmarks, v0.41.0 (inactive)
bracket-matcher, v0.82.1 (inactive)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (inactive)
dev-live-reload, v0.47.0 (inactive)
encoding-selector, v0.22.0 (inactive)
exception-reporting, v0.39.0 (inactive)
fuzzy-finder, v1.3.0 (inactive)
git-diff, v1.1.0 (inactive)
find-and-replace, v0.198.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (inactive)
image-view, v0.58.0 (inactive)
incompatible-packages, v0.26.1 (inactive)
keybinding-resolver, v0.35.0 (inactive)
line-ending-selector, v0.5.0 (inactive)
link, v0.31.1 (inactive)
markdown-preview, v0.158.1 (inactive)
metrics, v0.53.1 (inactive)
notifications, v0.64.1 (inactive)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.238.2 (inactive)
snippets, v1.0.2 (inactive)
spell-check, v0.67.1 (inactive)
status-bar, v1.4.1 (inactive)
styleguide, v0.47.1 (inactive)
symbols-view, v0.113.0 (inactive)
tabs, v0.98.1 (inactive)
timecop, v0.33.1 (inactive)
tree-view, v0.208.0 (inactive)
update-package-dependencies, v0.10.0 (inactive)
welcome, v0.34.0 (inactive)
whitespace, v0.32.2 (inactive)
wrap-guide, v0.38.1 (inactive)
language-c, v0.52.1 (inactive)
language-clojure, v0.21.0 (inactive)
language-coffee-script, v0.47.0 (inactive)
language-csharp, v0.12.1 (inactive)
language-css, v0.36.2 (inactive)
language-gfm, v0.86.0 (inactive)
language-git, v0.13.0 (inactive)
language-go, v0.42.0 (inactive)
language-html, v0.44.1 (inactive)
language-hyperlink, v0.16.0 (inactive)
language-java, v0.22.0 (inactive)
language-javascript, v0.119.0 (inactive)
language-json, v0.18.0 (inactive)
language-less, v0.29.3 (inactive)
language-make, v0.22.2 (inactive)
language-mustache, v0.13.0 (inactive)
language-objective-c, v0.15.1 (inactive)
language-perl, v0.35.0 (inactive)
language-php, v0.37.0 (inactive)
language-property-list, v0.8.0 (inactive)
language-python, v0.45.0 (inactive)
language-ruby, v0.68.5 (inactive)
language-ruby-on-rails, v0.25.0 (inactive)
language-sass, v0.52.0 (inactive)
language-shellscript, v0.22.3 (inactive)
language-source, v0.9.0 (inactive)
language-sql, v0.21.1 (inactive)
language-text, v0.7.1 (inactive)
language-todo, v0.28.0 (inactive)
language-toml, v0.18.0 (inactive)
language-xml, v0.34.8 (inactive)
language-yaml, v0.26.0 (inactive)

# Dev
No dev packages
```

**Comment by ddenisenko** [comment]
Could you please provide RAML example to reproduce the issue?

**Comment by shcherbak** [comment]
Hello, Denis!

I have no copy of theRAML file, but can remember that error was coused by
including complex  json-schema. error was on json validation.

examples of json attached

2016-09-16 7:34 GMT+03:00 Denis Denisenko <notifications@github.com>:

**Comment by ddenisenko** [comment]
Thank you, but it looks like JSON schema example attachment did not get it in through email. At least, I cant see it. Could you please attach it directly via github site?

**Comment by shcherbak** [comment]
here below is a json-schemas, caused the error

`{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "Warehouse Issues Collection",
  "description": "Warehouse Issues Collection Type definition",
  "type": "array",
  "items": {
    "$ref": "warehouse-issue.schema"
  },
  "uniqueItems": true
}`

`{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "Goods Issue Note",
  "description": "A Goods Issue is defined as a physical outbound movement of goods or materials from the warehouse or it is the issue of physical goods or materials from the warehouse. It results in decrease in stock from the warehouse. For e.g goods are issued from the store to production department to make the product. Kinds: Issues to a cost center or project; Material staging for production; Delivery of goods to customers",
  "type": "object",
  "properties": {
    "id": {
      "description": "The unique identifier for a document",
      "type": "integer"
    },
    "code": {
      "description": "The unique autogenerated code of a document",
      "type": "string",
      "pattern": "GIN-[0-9]{3}-[0-9]{6}"
    },
    "creation_date": {
      "description": "The date when the document is due",
      "type": "string",
      "pattern": "20[0-9]{2}-[0-1][0-9]-[0-3][0-9]"
    },
    "issued_by": {
      "description": "Issued by",
      "type": "array",
      "items": {
        "$ref": "contractor.schema"
      },
      "minItems": 1,
      "maxItems": 1
    },
    "received_by": {
      "description": "Received by",
      "type": "array",
      "items": {
        "$ref": "contractor.schema"
      },
      "minItems": 1,
      "maxItems": 1
    },
    "comment": {
      "description": "The comment that should be visible on the document",
      "type": "string"
    },
    "electronic_document": {
      "description": "1 if Issue needs digital signature and trusted timestamp",
      "type": "integer",
      "enum": [
        0,
        1
      ],
      "default": 1
    },
    "template_lang_code": {
      "description": "The language code",
      "type": "string",
      "enum": [
        "uk",
        "ru",
        "en"
      ]
    },
    "purpose": {
      "description": "Issued for purpose",
      "type": "string",
      "enum": [
        "internal",
        "manufacturing",
        "sales"
      ]
    },
    "type": {
      "description": "Issue type",
      "type": "integer",
      "enum": [
        0,
        1,
        2
      ]
    },
    "print_url": {
      "description": "An url for pdf downloading",
      "type": "string"
    },
    "content": {
      "description": "List of the items",
      "type": "array",
      "items": {
        "$ref": "goods-item.schema"
      },
      "minItems": 1,
      "uniqueItems": true
    }
  },
  "additionalProperties": false,
  "required": [
    "id",
    "content"
  ]
}`

**Comment by ddenisenko** [comment]
Looks working with these schemas, no exceptions.

**Comment by shcherbak** [comment]
this is a final version of json-schema. error occures during edition
process of included file

2016-09-22 7:48 GMT+03:00 Denis Denisenko <notifications@github.com>:

**Comment by ddenisenko** [comment]
>this is a final version of json-schema. error occures during edition
process of included file

Do you mean the previous schemas you posted in a comment, or there is an attachment missing?
The ones posted in the comment work for me. 

Would you mind attaching RAML file and the schemas as an archive (zip) and letting me know, which exactly file you are editing and what you change in the file when the error appears?

**Comment by KonstantinSviridov** [comment]
Hi, @shcherbak 

The exception can be observed after removing the first opening curly bracket.
The fix will be released in 0.8.40

Regards, Konstantin

**Comment by shcherbak** [comment]
Hi, Konstantin Sviridov

thank a lot

2016-10-25 14:17 GMT+03:00 Konstantin Sviridov <notifications@github.com>:
